### PR TITLE
Cleanup store

### LINKED ISSUE #407: Refactor mobx store init so constructor has no side effects
* Invoke computed modules in init
* Pull AsyncStorage read into `restore` function
* Use `new Store()` in unit tests

### PR DESCRIPTION
**Pull request by tanx**
Closes #407

### CODE REVIEW

File: test/unit/action/info.spec.js [hunk 1]
```diff
@@ -15,8 +15,7 @@ describe('Action Info Unit Tests', () => {
   beforeEach(() => {
     sandbox = sinon.createSandbox({});
     sandbox.stub(logger);
-    useStrict(false);
-    store = observable({ lndReady: false });
+    store = new Store();
```
Comment (valentinewallace) [inline test/unit/action/info.spec.js]: This refactor is very easy on the eyes :100: makes test writing way easier as well

File: src/action/index.js [hunk 2]
```diff
@@ -20,7 +20,8 @@ const ipcRenderer = window.ipcRenderer; // exposed to sandbox via preload.js
 // Inject dependencies
 //
 
-store.init(AsyncStorage);
+store.init();
```
Comment (valentinewallace) [inline src/action/index.js]: This breaking-up of functions makes sense to me

File: src/store.js [hunk 3]
```diff
@@ -33,32 +33,32 @@ export class Store {
         passwordVerify: '',
         seedVerify: ['', '', ''],
       },
-      transactions: null,
+      transactions: [],
```
Comment (valentinewallace) [inline src/store.js]: these look great! interesting we weren't initializing with empty lists from the beginning :O

**Comment by tanx** [comment]
Awesome. Thanks for the feedback.

**Pull request merged**